storage/memory.go

```diff
@@ -515,41 +515,25 @@ func (s *MemoryStore) DeletePARSession(ctx context.Context, requestURI string) (
 	return nil
 }
 
-// CreateDeviceCodeSession stores the device code session
-func (s *MemoryStore) CreateDeviceCodeSession(_ context.Context, signature string, req fosite.Requester) error {
-	// We first lock accessTokenRequestIDsMutex and then accessTokensMutex because this is the same order
-	// locking happens in RevokeAccessToken and using the same order prevents deadlocks.
-	s.deviceCodesRequestIDsMutex.Lock()
-	defer s.deviceCodesRequestIDsMutex.Unlock()
-	s.deviceCodesMutex.Lock()
-	defer s.deviceCodesMutex.Unlock()
+// CreateDeviceAuthSession stores the device auth session
+func (s *MemoryStore) CreateDeviceAuthSession(_ context.Context, deviceCodeSignature, userCodeSignature string, req fosite.Requester) error {
+	s.deviceAuthsRequestIDsMutex.Lock()
+	defer s.deviceAuthsRequestIDsMutex.Unlock()
+	s.deviceAuthsMutex.Lock()
+	defer s.deviceAuthsMutex.Unlock()
 
-	s.DeviceCodes[signature] = req
-	s.DeviceCodesRequestIDs[req.GetID()] = signature
-	return nil
-}
-
-// UpdateDeviceCodeSession updates the device code session
-func (s *MemoryStore) UpdateDeviceCodeSession(_ context.Context, signature string, req fosite.Requester) error {
-	s.deviceCodesRequestIDsMutex.Lock()
-	defer s.deviceCodesRequestIDsMutex.Unlock()
-	s.deviceCodesMutex.Lock()
-	defer s.deviceCodesMutex.Unlock()
-
-	// Only update if exist
-	if _, exists := s.DeviceCodes[signature]; exists {
-		s.DeviceCodes[signature] = req
-		s.DeviceCodesRequestIDs[req.GetID()] = signature
-	}
+	s.DeviceAuths[deviceCodeSignature] = req
+	s.DeviceAuths[userCodeSignature] = req
+	s.DeviceCodesRequestIDs[req.GetID()] = DeviceAuthPair{deviceCodeSignature, userCodeSignature}
```

Comment:
might be wrong but my linter used to glow when using assignments like this without specifying the name of the attribute